instancetype: Add basic expand-spec docs (#779)
kubevirt/user-guide: docs/virtual_machines/instancetypes.md

}
```

## expand-spec

> A VM subresource API and `virtctl` subcommand are provided to allow users to obtain a copy of their existing or yet to be defined VM with any referenced instance types or preferences expanded within the core spec of the VM. 

Reviewer: ```suggestion
The VM `expand-spec` subresource API and `virtctl expand` subcommand allow users to view an existing or yet-to-be defined VM with any referenced instance types or preferences expanded within the core spec of the VM. 
```
It took me a few reads to understand this so my suggestion may have changed the meaning if I've misunderstood.

    spec:
      domain:
        devices:
          interfaces:
          - masquerade: {}
            name: default
        machine:
        resources: {}
      networks:
      - name: default
        pod: {}
      volumes:
      - containerDisk:
          image: registry:5000/kubevirt/cirros-container-disk-demo:devel
        name: cirros
EOF
```

### virtctl expand

```yaml
$ diff -u <(kubectl get vm/cirros -o yaml | yq .spec) <(virtctl expand --vm cirros -o yaml | yq .spec)
--- /dev/fd/63  2024-03-25 13:37:27.948090870 +0000
+++ /dev/fd/62  2024-03-25 13:37:27.948090870 +0000
@@ -1,7 +1,3 @@
-instancetype:
-  kind: virtualmachineclusterinstancetype
-  name: u1.micro
-  revisionName: cirros-u1.micro-1b23a27b-a7f2-4d69-8b3d-ee609a99d4a3-1
 runStrategy: Always
 template:
   metadata:
@@ -9,12 +5,18 @@
   spec:
     architecture: amd64
     domain:
+      cpu:
+        cores: 1
+        sockets: 1
+        threads: 1
       devices:
         interfaces:
           - masquerade: {}
             name: default
       machine:
         type: q35
+      memory:
+        guest: 1Gi
       resources: {}
     networks:
       - name: default
```

### subresources.kubevirt.io expand-spec

> 

Reviewer: Similarly it would be nice to have a brief explanation that this is getting the same result as above but through use of the API. It might be good to change the title as well to something like "Example for using the `expand-spec` API subresource"
